register.go

```diff
@@ -127,33 +239,51 @@ func unregister() error {
 		return fmt.Errorf("warning: the system is already unregistered")
 	}
 
-	if err := conn.Object("com.redhat.RHSM1", "/com/redhat/RHSM1/Unregister").Call("com.redhat.RHSM1.Unregister.Unregister", dbus.Flags(0), map[string]string{}, "").Err; err != nil {
-		return unpackError(err)
+	if err := conn.Object(
+		"com.redhat.RHSM1",
+		"/com/redhat/RHSM1/Unregister").Call(
+		"com.redhat.RHSM1.Unregister.Unregister",
+		dbus.Flags(0),
+		map[string]string{},
+		"").Err; err != nil {
+		return unpackRHSMErrorMsg(err)
 	}
 
 	return nil
 }
 
-func unpackError(err error) error {
+// RHSMError is used for parsing JSON document returned by D-Bus methods.
+type RHSMError struct {
+	Exception string `json:"exception"`
+	Severity  string `json:"severity"`
+	Message   string `json:"message"`
+}
+
+// unpackRHSMError tries to unpack JSON document (part of error) into the structure RHSMError. When it is
+// not possible to parse error into structure, then return nil pointer and error. When it is
+// possible to parse error into structure, then returned pointer on structure contains valid data and err is nil.
+func unpackRHSMError(err error) (*RHSMError, error) {
```

Comment:
I'm having trouble trying to figure out the difference between `unpackRHSMError` and `unpackRHSMErrorMsg`. They seem to be doing mostly the same thing. Maybe `unpackRHSMErrorMsg` needs to be moved to a method on the `RHSMError` struct?

```go
func (r RHSMError) Error() string {
    return fmt.Sprintf("%v: %v", r.Severity, r.Message)
}
```

This has the added advantage of implementing the Error interface, allowing an RHSMError value to be used as an `error` type.

Reply:
Function `unpackRHSMErrorMsg()` is needed to get error message in the most cases, but it is necessary to unpack only the error (and use `unpackRHSMError()`) in `registerUsernamePassword()`, because it is necessary to compare `rhsmError.Exception` with `""OrgNotSpecifiedException"`.

I can move `unpackRHSMErrorMsg` as a method of `RHSMError` struct.

Reply:
Yea, I can see they both have a purpose, but it's not clear what each function's purpose is. Since `unpackRHSMErrorMsg` seems to always require an `RHSMError` value, it effectively is a method on that type. Structuring things that way should make each method's intended purpose more explicit.